packages/keystone/databaseAdapters/ReplicaKnexAdapter.js

```diff
@@ -54,6 +57,18 @@ class ReplicaKnexAdapter extends KnexAdapter {
 
         //override runner method
         this.knex.client.runner = (builder) => {
+            const graphqlContext = graphqlCtx.getStore()
+
+            const operationType = get(graphqlContext, ['info', 'operation', 'operation'])
+
+            // NOTE: any DB calls inside mutation => master, any other calls routed by SQL operation.
+            // Probably we want to pass function (builder => { replicaSet: string, master: boolean }) later in constructor,
+            // but for now "hard-coded" solution seems fine
+            if (operationType === 'mutation') {
+                return this.knexWrite.client.runner(builder)
+            }
+
+            // TODO(SavelevMatthew): think about removing this block later
```

Comment:
You should also remember about transactions, such as migrations, index creations and etc

Reply:
Yep, I'm talking about `queryContext.useMaster`. If I've understood its purpose correctly, then its goal is to route request based on JS-flag. So, block above does the same. Right now I left it as is, since I'm not sure (more routing = good or bad), but we'll figure it out later.

I'm also not planning to remove`checkUseMasterSingle`, since its goal to parse raw SQL (such as migrations / indexes and so on), not business (JS) logic. So we should keep it 100%

Reply:
you should add tasknumber here